packages/web/components/input/dynamic-size-input.tsx

```diff
@@ -0,0 +1,55 @@
+import classNames from "classnames";
+import { FC, useCallback, useEffect, useRef, useState } from "react";
+
+export interface DynamicSizeInputProps<T = string> {
+  value: T;
+  onChange: (newValue: T) => void;
+  className?: string;
+  type: "number" | "string";
+  placeholder?: string;
+  disabled?: boolean;
+  onClick?: () => void;
+}
+
+export const DynamicSizeInput: FC<DynamicSizeInputProps> = ({
```

Comment:
BTW I think we import a `react-input-autosize` package that may do something like this